sssd_test_framework/roles/generic.py

```diff
@@ -127,6 +129,35 @@ def fqn(self, name: str) -> str:
         """
         pass
 
+    @abstractmethod
+    def dns(self) -> GenericDNSServer:
+        """
+        Get DNS server object.
+
+            Get methods use dig and is parsed by jc. The data from jc contains several nested dict,
+            but two are returned as a tuple, ``answer, authority``.
+
+        .. code-block:: python
+            :caption: Example usage
+
+            # Create forward zone, add and get record and assert ip
+            forward_zone = provider.dns().zone("smb.test").create()
+            forward_zone.add_record("client", "172.16.200.5")
+            result = forward_zone.get_record("client.smb.test")
+            assert result[0].get("data") == "172.16.200.5"
+
+            # Create reverse zone, add and get record and assert ttl
+            reverse_zone = provider.dns().zone("10.0.10.in-addr.arpa").create()
+            reverse_zone.add_ptr_record("client.provider.test", 15)
+
+            result = reverse_zone.get_ptr_record("10.0.10.15")
+            assert "3600" == result[0].get("ttl"), "ttl does not equal 3600!"
+
+            # Assert if the record exists
+            assert reverse_zone.ptr_record_exists("10.0.10.15"), "ptr record does not exist!"
```

Comment:
Why is `ptr_record_exists` needed? can't we just use `result = reverse_zone.get_ptr_record("10.0.10.15")` and assert result is none?

Reply:
Okay I understand, i'm fine to keep it if you think it will be of use.